diff --git a/node_modules/@remix-run/react/dist/esm/routes.js b/node_modules/@remix-run/react/dist/esm/routes.js
index c06ab4f..d5d805c 100644
--- a/node_modules/@remix-run/react/dist/esm/routes.js
+++ b/node_modules/@remix-run/react/dist/esm/routes.js
@@ -150,6 +150,10 @@ function getRedirect(response) {
   if (revalidate) {
     headers["X-Remix-Revalidate"] = revalidate;
   }
+  let reloadDocument = response.headers.get("X-Remix-Reload-Document");
+  if (reloadDocument) {
+    headers["X-Remix-Reload-Document"] = reloadDocument;
+  }
   return redirect(url, {
     status,
     headers
diff --git a/node_modules/@remix-run/react/dist/routes.js b/node_modules/@remix-run/react/dist/routes.js
index e2a97f6..15f4821 100644
--- a/node_modules/@remix-run/react/dist/routes.js
+++ b/node_modules/@remix-run/react/dist/routes.js
@@ -171,6 +171,10 @@ function getRedirect(response) {
   if (revalidate) {
     headers["X-Remix-Revalidate"] = revalidate;
   }
+  let reloadDocument = response.headers.get("X-Remix-Reload-Document");
+  if (reloadDocument) {
+    headers["X-Remix-Reload-Document"] = reloadDocument;
+  }
   return reactRouterDom.redirect(url, {
     status,
     headers
