diff --git a/node_modules/@remix-run/react/dist/esm/routes.js b/node_modules/@remix-run/react/dist/esm/routes.js
index f1a873a..5c2d6bf 100644
--- a/node_modules/@remix-run/react/dist/esm/routes.js
+++ b/node_modules/@remix-run/react/dist/esm/routes.js
@@ -121,6 +121,10 @@ function getRedirect(response) {
   if (revalidate) {
     headers["X-Remix-Revalidate"] = revalidate;
   }
+  let hardRedirect = response.headers.get("X-Remix-Hard-Redirect");
+  if (hardRedirect) {
+    headers["X-Remix-Hard-Redirect"] = hardRedirect;
+  }
   return redirect(url, {
     status,
     headers
diff --git a/node_modules/@remix-run/react/dist/routes.js b/node_modules/@remix-run/react/dist/routes.js
index c4bd0b1..d498d1a 100644
--- a/node_modules/@remix-run/react/dist/routes.js
+++ b/node_modules/@remix-run/react/dist/routes.js
@@ -145,6 +145,10 @@ function getRedirect(response) {
   if (revalidate) {
     headers["X-Remix-Revalidate"] = revalidate;
   }
+  let hardRedirect = response.headers.get("X-Remix-Hard-Redirect");
+  if (hardRedirect) {
+    headers["X-Remix-Hard-Redirect"] = hardRedirect;
+  }
   return reactRouterDom.redirect(url, {
     status,
     headers
